src/libopensc/card-starcos.c

```diff
@@ -1944,18 +2087,14 @@ static int starcos_get_serialnr(sc_card_t *card, sc_serial_number_t *serial)
 		return SC_SUCCESS;
 	}
 
-	switch (card->type) {
-		case SC_CARD_TYPE_STARCOS_V3_4:
-		case SC_CARD_TYPE_STARCOS_V3_5:
+	if ( IS_V3x(card) ) {
 			card->serialnr.len = SC_MAX_SERIALNR;
 			r = sc_parse_ef_gdo(card, card->serialnr.value, &card->serialnr.len, NULL, 0);
 			if (r < 0) {
 				card->serialnr.len = 0;
 				return r;
 			}
-			break;
-
-		default:
+	} else {
 			/* get serial number via GET CARD DATA */
 			sc_format_apdu(card, &apdu, SC_APDU_CASE_2_SHORT, 0xf6, 0x00, 0x00);
 			apdu.cla |= 0x80;
```

Comment:
After conversion from `switch` to `if`, the block contents are indented too much

Reply:
I will fix it.